mix.exs

```diff
@@ -0,0 +1,27 @@
+defmodule HookLineAndSlacker.Mixfile do
+  use Mix.Project
+
+  def project do
+    [app: :hook_line_and_slacker,
+     version: "0.1.0",
+     elixir: "~> 1.4",
+     build_embedded: Mix.env == :prod,
+     start_permanent: Mix.env == :prod,
+     deps: deps()]
+  end
+
+  def application do
+    [extra_applications: [:logger],
+     mod: {HookLineAndSlacker, []}]
+  end
+
+  defp deps do
+    [
+      {:cowboy, "~> 1.1"},
+      {:plug, "~> 1.3"},
+      {:poison, "~> 3.0"},
+      {:slackex, "~> 0.0.1"},
```

Comment:
Slackex is one of two slack libraries available in Elixir. The [other one](https://github.com/BlakeWilliams/Elixir-Slack) is clearly far too advanced to be necessary in this project.

Slackex was borderline. I chose to use it because I figured it would save me some time and it isn't a large dependency.

Of course this task could quite easily be accomplished with a raw POST as well, especially if you were absolutely sure that you would only ever be implementing this one webhook. If you were looking to minimize dependencies, that would be a reasonable choice IMO.

If you were implementing (or intended to implement) more than one kind of interaction with Slack, I think using an external and well-tested library makes sense.